pkg/controller/component/available.go

```diff
@@ -0,0 +1,474 @@
+		return h.evaluateAll(*cond.All, replicas, events), "", nil
+	}
+	if cond.Any != nil {
+		return h.evaluateAny(*cond.Any, replicas, events), "", nil
+	}
+	if cond.None != nil {
+		return h.evaluateNone(*cond.None, replicas, events), "", nil
+	}
+	if cond.Majority != nil {
+		return h.evaluateMajority(*cond.Majority, replicas, events), "", nil
+	}
+	return true, "", nil
+}
+
+func (h *AvailableEventHandler) evaluateAnd(conditions []appsv1.ComponentAvailableConditionX, replicas int32, events []probeEvent) bool {
+	for _, cond := range conditions {
+		ok, _, _ := h.evaluateConditionX(cond, replicas, events)
+		if !ok {
+			return false
+		}
+	}
+	return true
+}
+
+func (h *AvailableEventHandler) evaluateOr(conditions []appsv1.ComponentAvailableConditionX, replicas int32, events []probeEvent) bool {
+	for _, cond := range conditions {
+		ok, _, _ := h.evaluateConditionX(cond, replicas, events)
+		if ok {
+			return true
+		}
+	}
+	return false
+}
+
+func (h *AvailableEventHandler) evaluateNot(cond appsv1.ComponentAvailableConditionX, replicas int32, events []probeEvent) bool {
+	ok, _, _ := h.evaluateConditionX(cond, replicas, events)
+	return !ok
+}
+
+func (h *AvailableEventHandler) evaluateAll(cond appsv1.ComponentAvailableConditionX, replicas int32, events []probeEvent) bool {
+	if !h.strictCheck(cond, replicas, events) {
+		return false
+	}
+	for _, event := range events {
+		ok, _, _ := h.evaluateConditionX(cond, replicas, []probeEvent{event})
+		if !ok {
+			return false
+		}
+	}
+	return true
+}
+
+func (h *AvailableEventHandler) evaluateAny(cond appsv1.ComponentAvailableConditionX, replicas int32, events []probeEvent) bool {
+	if !h.strictCheck(cond, replicas, events) {
+		return false
+	}
+	for _, event := range events {
+		ok, _, _ := h.evaluateConditionX(cond, replicas, []probeEvent{event})
+		if ok {
+			return true
+		}
+	}
+	return false
+}
+
+func (h *AvailableEventHandler) evaluateNone(cond appsv1.ComponentAvailableConditionX, replicas int32, events []probeEvent) bool {
+	if !h.strictCheck(cond, replicas, events) {
+		return false
+	}
+	for _, event := range events {
+		ok, _, _ := h.evaluateConditionX(cond, replicas, []probeEvent{event})
+		if ok {
+			return false
+		}
+	}
+	return true
+}
+
+func (h *AvailableEventHandler) evaluateMajority(cond appsv1.ComponentAvailableConditionX, replicas int32, events []probeEvent) bool {
+	count := 0
+	for _, event := range events {
+		ok, _, _ := h.evaluateConditionX(cond, replicas, []probeEvent{event})
+		if ok {
+			count++
+		}
+	}
+	return int32(count) > replicas/2
+}
+
+func (h *AvailableEventHandler) strictCheck(cond appsv1.ComponentAvailableConditionX, replicas int32, events []probeEvent) bool {
+	if cond.Strict != nil && *cond.Strict {
+		if replicas != int32(len(events)) {
+			return false
+		}
+	}
+	return true
+}
+
+func (h *AvailableEventHandler) evaluateConditionX(cond appsv1.ComponentAvailableConditionX, replicas int32, events []probeEvent) (bool, string, error) {
```

Comment:
The last two returned values are always empty, maybe just remove them.

Reply:
The returned error has been deleted, and the message is retained to convey detailed information when it is unavailable. It will be added later.